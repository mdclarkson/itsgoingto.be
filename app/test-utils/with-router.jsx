import React from 'react'
import PropTypes from 'prop-types'
import { BrowserRouter as Router } from 'react-router-dom'

const WithRouter = ({ children }) => (
  <Router>
    {children}
  </Router>
)

WithRouter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
}

export default WithRouter
