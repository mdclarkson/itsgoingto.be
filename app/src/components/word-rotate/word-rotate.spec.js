import React from 'react'
import { render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import WordRotate from './word-rotate'

describe('(Component) WordRotate', () => {
  describe('(Render)', () => {
    jest.useFakeTimers()
    let renderResult

    beforeEach(() => {
      renderResult = render(<WordRotate words="word1,word2" />)
    })

    afterEach(() => {
      jest.clearAllTimers()
    })

    describe('initial word (first word)', () => {
      it('matches snapshot', () => {
        const { asFragment } = renderResult

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('after interval passed (second word)', () => {
      it('matches snapshot', () => {
        const { asFragment } = renderResult

        act(() => {
          jest.advanceTimersByTime(5000)
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('after interval passed (back to first word)', () => {
      it('matches snapshot', () => {
        const { asFragment } = renderResult

        act(() => {
          jest.advanceTimersByTime(10000)
        })

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
