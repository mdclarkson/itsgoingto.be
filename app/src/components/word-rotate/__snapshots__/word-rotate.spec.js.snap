// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) WordRotate (Render) after interval passed (back to first word) matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="word-rotate"
  >
    <span
      class="word-rotate_word"
    >
      word1
    </span>
    <span
      class="word-rotate_word word-rotate_word--previous"
    >
      word2
    </span>
  </span>
</DocumentFragment>
`;

exports[`(Component) WordRotate (Render) after interval passed (second word) matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="word-rotate"
  >
    <span
      class="word-rotate_word"
    >
      word2
    </span>
    <span
      class="word-rotate_word word-rotate_word--previous"
    >
      word1
    </span>
  </span>
</DocumentFragment>
`;

exports[`(Component) WordRotate (Render) initial word (first word) matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="word-rotate"
  >
    <span
      class="word-rotate_word"
    >
      word1
    </span>
    <span
      class="word-rotate_word word-rotate_word--previous"
    >
      word2
    </span>
  </span>
</DocumentFragment>
`;
